xray/conventions.py

```diff
@@ -215,8 +215,6 @@ def encode_cf_datetime(dates, units=None, calendar=None):
     --------
     netCDF4.date2num
     """
-    import netCDF4 as nc4
```

Comment:
only if decoding without netCDF4 fails.

Probably would be cleaner to separate this into two sub-functions